plugins/modules/ndo_dhcp_option_policy.py

```diff
@@ -172,23 +172,17 @@ def main():
     mso_template.validate_template("tenantPolicy")
 
     path = "/tenantPolicyTemplate/template/dhcpOptionPolicies"
-    existing_dhcp_option_policies = mso_template.template.get("tenantPolicyTemplate", {}).get("template", {}).get("dhcpOptionPolicies", [])
-    if option_policy:
-        object_description = "DHCP Option Policy"
-        if option_policy_uuid:
-            match = mso_template.get_object_by_uuid(object_description, existing_dhcp_option_policies, option_policy_uuid)
-        else:
-            kv_list = [KVPair("name", option_policy)]
-            match = mso_template.get_object_by_key_value_pairs(object_description, existing_dhcp_option_policies, kv_list)
-        if match:
-            mso.existing = mso.previous = copy.deepcopy(match.details)
-    else:
-        mso.existing = mso.previous = existing_dhcp_option_policies
 
-    if state == "present":
+    match = get_dhcp_option_policy(mso_template, option_policy_uuid, option_policy)
 
+    if option_policy_uuid or option_policy:
         if match:
+            mso.existing = mso.previous = copy.deepcopy(match.details)  # Query a specific object
+    elif match:
+        mso.existing = match  # Query all objects
 
+    if state == "present":
+        if match:
```

Comment:
are we not using the append ops function here?

Reply:
The append ops function will be handled with a separate issue.